mailpoet/lib/Segments/DynamicSegments/Filters/WooCommercePurchaseDate.php

```diff
@@ -32,20 +32,20 @@ public function apply(QueryBuilder $queryBuilder, DynamicSegmentFilterEntity $fi
     $operator = $this->dateFilterHelper->getOperatorFromFilter($filter);
     $dateValue = $this->dateFilterHelper->getDateValueFromFilter($filter);
     $date = $this->dateFilterHelper->getDateStringForOperator($operator, $dateValue);
-    $subQuery = $this->getSubQuery($operator, $date);
     $subscribersTable = $this->filterHelper->getSubscribersTable();
 
     if (in_array($operator, [DateFilterHelper::NOT_ON, DateFilterHelper::NOT_IN_THE_LAST])) {
+      $subQuery = $this->filterHelper->getNewSubscribersQueryBuilder();
+      $this->applyConditionsToQueryBuilder($operator, $date, $subQuery);
```

Comment:
It's already late in the day here, so I might not be fully getting this, but would changing [the conditions in the switch](https://github.com/mailpoet/mailpoet/pull/4760/files#diff-2f21b766853eb8be74b385915052c5176352b93e3cdc8d457e3d05607cebaa6eR52) help?

What if we change this:
```php
case DateFilterHelper::IN_THE_LAST:
case DateFilterHelper::NOT_IN_THE_LAST:
  $queryBuilder->andWhere("DATE($orderStatsAlias.date_created) >= $quotedDate");
  break;
case DateFilterHelper::ON:
case DateFilterHelper::NOT_ON:
  $queryBuilder->andWhere("DATE($orderStatsAlias.date_created) = $quotedDate");
```

To sth like this:

```php
case DateFilterHelper::IN_THE_LAST:
  $queryBuilder->andWhere("DATE($orderStatsAlias.date_created) < $quotedDate");
  break;
case DateFilterHelper::NOT_IN_THE_LAST:
  $queryBuilder->andWhere("DATE($orderStatsAlias.date_created) >= $quotedDate");
  break;
case DateFilterHelper::ON:
  $queryBuilder->andWhere("DATE($orderStatsAlias.date_created) != $quotedDate");
  break;
case DateFilterHelper::NOT_ON:
  $queryBuilder->andWhere("DATE($orderStatsAlias.date_created) = $quotedDate");
  break;
```

They're inner joins, so... wouldn't it work?